### DISCUSSION TITLE
Updating arrays within context via assign

### DISCUSSION THREAD

**Discussion by dsathyakumar** [statelyai/xstate#2088] [Q&A] [answered]
Hello 👋 

if my context object had an array like 

```
{
   ....
    context : {
        data: []
    }
    ....
}
```

How can new items be pushed into this via the `assign` action.
Most of the examples seem to return a new object to the props within context. 
I just want to be able to push an item into the `data` array. 
Is this possible with `assign` ?

**Comment by davidkpiano** [comment, marked as answer]
You should do it immutably. Use `.concat(...)` instead.

**Reply by dsathyakumar** [reply]
Thanks 👍 But .concat() creates a new `[]` 🤔

**Reply by davidkpiano** [reply]
That's right; your updates in `assign()` should always be pure.
